src/Symfony/Component/Messenger/TraceableMessageBus.php

```diff
@@ -40,7 +42,7 @@ public function dispatch($message): Envelope
         ];
 
         try {
-            return $this->decoratedBus->dispatch($message);
+            return $this->decoratedBus->dispatch($message, ...$stamps);
```

Comment:
I think the extra stamps should be part of $context

Reply:
maybe we should replace `$message instanceof Envelope ? $message : new Envelope($message);`
by `$envelope = Envelope::wrap($message, $stamps);` everywhere?